Source/CombatExtended/CombatExtended/Projectiles/BulletCE.cs

```diff
@@ -168,14 +164,19 @@ private void NotifyImpact(Thing hitThing, Map map, IntVec3 position)
         /* Used for creating instances of Bullet for use with Thing.Notify_BulletImpactNearby.
          * Current users are SmokepopBelt and BroadshieldPack, requiring bullet.def and bullet.Launcher.
          */
+
+        // todo: remove when moved to publicised assembly
+        private static readonly FieldInfo bulletLauncher = typeof(Bullet).GetField("launcher", BindingFlags.Instance | BindingFlags.NonPublic);
+
         private Bullet GenerateVanillaBullet()
         {
             var bullet = new Bullet
             {
                 def = this.def,
                 intendedTarget = this.intendedTarget,
             };
-            Traverse.Create(bullet).Field("launcher").SetValue(this.launcher);  //Bad for performance, refactor if a more efficient solution is possible
+
+            bulletLauncher.SetValue(bullet, this.launcher);  //Bad for performance, refactor if a more efficient solution is possible
```

Comment:
Is this still bad for performance?

Reply:
Yes, as I understand it, a traverse is just a wrapper for reading/writing fields of an object, and calls SetValue itself, it just caches the retrieval of the fieldinfo (which I also did by making it a static field) see [retrieving from cache](https://github.com/pardeike/Harmony/blob/master/Harmony/Tools/Traverse.cs#L247),  [setting the value](https://github.com/pardeike/Harmony/blob/master/Harmony/Tools/Traverse.cs#L185-L194)


I just changed it because while I may not improve the retrieval side of it, I can at the very least remove the creation of the traverse object (I believe the lookup for the field is negligible).